packages/renderer/src/lib/markdown/micromark-listener-handler.ts

```diff
@@ -76,44 +67,47 @@ export function createListener(
       }
     }
 
+    // BUTTON CHECK: 'expandable'
     // if the user clicked on the toggle of an expandable section
     if (expandable) {
       executeExpandableToggle(expandable);
       return;
     }
 
+    // BUTTON CHECK: 'generic' button click (not a command)
     // if the user clicked on a button (new way)
-    if (!command && eventTarget instanceof HTMLButtonElement) {
-      const targetId = eventTarget.id;
+    if (!command && e.target instanceof HTMLButtonElement) {
+      const targetId = e.target.id;
       executeButtonCommand(targetId).catch((err: unknown) => console.error(`Error executing command ${targetId}`, err));
       return;
     }
 
+    // BUTTON CHECK: 'command' button click (used in onboarding package.json)
     // Only check if the command exists and the target is not disabled
-    if (command && 'disabled' in eventTarget && !eventTarget.disabled) {
+    if (command) {
       // If the target is an instance of a button element, we know that we are going to execute either
       // a command or hyperlink
-      if (eventTarget instanceof HTMLButtonElement) {
-        const targetButton = eventTarget as HTMLButtonElement;
+      if (e.target instanceof HTMLButtonElement && !e.target.disabled) {
         // If the command exists and the button is not disabled, we execute the command
         // we'll also be updating the inProgressMarkdownCommandExecutionCallback so we have
         // real-time updates on the button
         inProgressMarkdownCommandExecutionCallback(command, 'starting');
-        targetButton.disabled = true;
-        if (targetButton.firstChild && targetButton.firstChild instanceof HTMLElement) {
-          targetButton.firstChild.style.display = 'inline-block';
+        e.target.disabled = true;
+        if (e.target instanceof HTMLElement && e.target.firstChild instanceof HTMLElement) {
+          e.target.firstChild.style.display = 'inline-block';
         }
+
         window
           .executeCommand(command, args)
           .then(value => inProgressMarkdownCommandExecutionCallback(command, 'successful', value))
           .catch((reason: unknown) => inProgressMarkdownCommandExecutionCallback(command, 'failed', reason))
           .finally(() => {
-            targetButton.disabled = false;
-            if (targetButton.firstChild && targetButton.firstChild instanceof HTMLElement) {
-              targetButton.firstChild.style.display = 'none';
+            (e.target as HTMLButtonElement).disabled = false;
+            if (e.target instanceof HTMLElement && e.target.firstChild instanceof HTMLElement) {
```

Comment:
HTMButtonElement inherited from HTMLElement. Considering it is already checked above (line 90) it is instance of HTMLButtonElement this instanceof is not required.

Reply:
Still need this in place, or I get a typescript linting error.